### PR TITLE
CI Workflow: Add all AWS supported k8s versions

### PR DESCRIPTION
**Pull request by brlbil**
The current awd conformance tests run against one version of k8s either a specific version or the default version that is provided by the cloud provider.

This commit adds all the supported k8s versions by AWS in a matrix strategy.

Normal test runs:
:x: [conformance-aws-cni-v1.11.yaml](https://github.com/cilium/cilium/actions/runs/5314147605), failure is not related, it has been failing for some time
:white_check_mark: [conformance-aws-cni-v1.12.yaml](https://github.com/cilium/cilium/actions/runs/5314147615)
:white_check_mark: [conformance-aws-cni-v1.13.yaml](https://github.com/cilium/cilium/actions/runs/5314147607)
:white_check_mark: [conformance-aws-cni.yaml](https://github.com/cilium/cilium/actions/runs/5314147595)
:white_check_mark: [conformance-eks-v1.11.yaml](https://github.com/cilium/cilium/actions/runs/5314147619)
:white_check_mark: [conformance-eks-v1.12.yaml](https://github.com/cilium/cilium/actions/runs/5314147602)
:white_check_mark: [conformance-eks-v1.13.yaml](https://github.com/cilium/cilium/actions/runs/5314147594)
:white_check_mark: [conformance-eks.yaml](https://github.com/cilium/cilium/actions/runs/5314147606)

Simulated scheduled runs:
:x: [conformance-aws-cni-v1.11.yaml](https://github.com/cilium/cilium/actions/runs/5314733901), failure is not related, it has been failing for some time
:white_check_mark: [conformance-aws-cni-v1.12.yaml](https://github.com/cilium/cilium/actions/runs/5314733887)
:white_check_mark: [conformance-aws-cni-v1.13.yaml](https://github.com/cilium/cilium/actions/runs/5314733899)
:white_check_mark: [conformance-aws-cni.yaml](https://github.com/cilium/cilium/actions/runs/5314733897)
:white_check_mark: [conformance-eks-v1.11.yaml](https://github.com/cilium/cilium/actions/runs/5320181541)
:white_check_mark: [conformance-eks-v1.12.yaml](https://github.com/cilium/cilium/actions/runs/5320948199)
:white_check_mark: [conformance-eks-v1.13.yaml](https://github.com/cilium/cilium/actions/runs/5321686572)
:white_check_mark: [conformance-eks.yaml](https://github.com/cilium/cilium/actions/runs/5322430607)

Depends on: #26356

### CODE REVIEW

File: .github/actions/aws/k8s-versions.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    region: eu-central-1
+  - version: "1.24"
+    region: ap-northeast-1
+  - version: "1.25"
+    region: us-east-2
+    default: true
+  - version: "1.26"
+    region: ca-central-1
+  - version: "1.27"
+    region: eu-north-1
```
Comment (nbusseneau) [inline .github/actions/aws/k8s-versions.yaml]: [Nit, non-blocking] Let's try to make sure all the files end with an empty line for consistency.

File: .github/workflows/conformance-eks.yaml [hunk 2]
```diff
@@ -333,35 +404,48 @@ jobs:
...
+      - uses: Sibz/github-status-action@650dd1a882a76dbbbc4576fb5974b8d22f29847f # v1.1.6
         with:
           authToken: ${{ secrets.GITHUB_TOKEN }}
-          sha: ${{ steps.vars.outputs.sha }}
+          sha: ${{ needs.setup-report.outputs.sha }}
           context: ${{ github.workflow }}
           description: Connectivity test cancelled
           state: error
           target_url: ${{ env.check_url }}
+
```
Comment (nbusseneau) [inline .github/workflows/conformance-eks.yaml]: [Nit] Opposite issue here 😅

**Review by nbusseneau — APPROVED** [review]
LGTM provided we address the dependency on #26356, some whitespace nits below (non-blocking).

**Review by bimmlerd — APPROVED** [review]
contributing lgtm

**Pull request merged**